.github/workflows/ci.yaml

```diff
@@ -26,38 +29,85 @@ jobs:
+        shell: bash -x -e {0}
+        run: |
+          echo "PUBLISH=${{ github.event_name == 'schedule' || (github.event_name == 'workflow_dispatch' && inputs.PUBLISH) }}" >> $GITHUB_OUTPUT
+
+      - name: Determine whether need to bump manifest 
+        id: if-bump-manifest
         shell: bash -x -e {0}
         run: |
-          if [[ -z "${{ inputs. TRIAL_BRANCH}}" ]]; then
-            echo "TRIAL_BRANCH=${{ github.sha }}"
-          else
-            echo "TRIAL_BRANCH=${TRIAL_BRANCH}"
-          fi | tee $GITHUB_OUTPUT
+          echo "BUMP_MANIFEST=${{ github.event_name == 'schedule' }}" >> $GITHUB_OUTPUT
 
   amd64:
     needs: metadata
     uses: ./.github/workflows/_ci.yaml
     with:
       ARCHITECTURE: amd64
-      TRIAL_BRANCH: ${{ needs.metadata.outputs.TRIAL_BRANCH }}
+      BUILD_DATE: ${{ needs.metadata.outputs.BUILD_DATE}}
+      BUMP_MANIFEST: ${{ needs.metadata.outputs.BUMP_MANIFEST}}
     secrets: inherit
  
   arm64:
     needs: metadata
     uses: ./.github/workflows/_ci.yaml
     with:
       ARCHITECTURE: arm64
-      TRIAL_BRANCH: ${{ needs.metadata.outputs.TRIAL_BRANCH }}
+      BUILD_DATE: ${{ needs.metadata.outputs.BUILD_DATE}}
+      BUMP_MANIFEST: ${{ needs.metadata.outputs.BUMP_MANIFEST}}      
     secrets: inherit
 
+  publish:
+    runs-on: ubuntu-22.04
+    needs: [metadata, amd64, arm64]
+    if: ${{ !cancelled() }} && needs.metadata.outputs.PUBLISH
+    strategy:
+      fail-fast: false
+      matrix:
+        TARGET_IMAGE: [jax, pallas, upstream-pax, upstream-t5x, pax, t5x]
+        FLAVOR: [mealkit, final]
+    steps:
+      - name: Transform matrix TARGET_IMAGE into docker container tag
+        id: transform
+        run: |
+          CONTAINER_TAG=$(echo ${{ matrix.TARGET_IMAGE }} | tr '[[:lower:]]' '[[:upper:]]' | tr '-' '_')
+          TARGET_TAG_1=${{ matrix.FLAVOR }}
+          TARGET_TAG_2=${{ matrix.FLAVOR }}
+          if [[ "${{ matrix.FLAVOR }}" == "final" ]]; then
+            TARGET_TAG_1="latest"
+            TARGET_TAG_2="nightly"
+          fi
+          echo "CONTAINER_TAG=${CONTAINER_TAG}" >> $GITHUB_OUTPUT
+          echo "TARGET_TAG_1=${TARGET_TAG_1}" >> $GITHUB_OUTPUT
+          echo "TARGET_TAG_2=${TARGET_TAG_2}" >> $GITHUB_OUTPUT
+      - name: Publish container
+        uses: ./.github/workflows/_publish_container.yaml
+        with:
+          SOURCE_IMAGE: |
+            ${{ fromJson(needs.amd64.outputs.ALL_TAGS)[format('TAG_{0}_{1}', steps.transform.outputs.CONTAINER_TAG, matrix.FLAVOR)] }}
+            ${{ fromJson(needs.arm64.outputs.ALL_TAGS)[format('TAG_{0}_{1}', steps.transform.outputs.CONTAINER_TAG, matrix.FLAVOR)] }}
+          TARGET_IMAGE: ${{ matrix.TARGET_IMAGE }}
+          TARGET_TAGS: |
+            type=raw,value=${{ steps.transform.outputs.TARGET_TAG_1 }},priority=500
+            type=raw,value=${{ steps.transform.outputs.TARGET_TAG_2 }}-${{ needs.metadata.outputs.BUILD_DATE }},priority=500
+
   finalize:
     needs: [amd64, arm64]
-    if: "!cancelled()"
```

Comment:
This change is not necessary. Do you have a particular reason?

Reply:
Reverted